geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/FeatureStateFactory.scala

```diff
@@ -260,4 +300,19 @@ object FeatureStateFactory extends LazyLogging {
 
     override def close(): Unit = executor.shutdownNow()
   }
+
+  class FilteredExpiryFactory(delegates: Seq[(Filter, FeatureStateFactory)]) extends FeatureStateFactory {
+
+    require(delegates.last._1 == Filter.INCLUDE,
```

Comment:
Ooo....  do the filters get reordered above?  

If I read correctly, I think the fallback is only added at the end if an INCLUDE is not present.  A user adding an INCLUDE->timeout1 and a second filter would hit and have an error:(.

Reply:
good point - i will add some validation to fail up front in that case (since anything after INCLUDE would never be invoked)

Reply:
Your review of my review comment was excellent.  POINTS!